src/consensus.rs

```diff
@@ -417,25 +435,37 @@ impl Consensus {
             .map(|(id, _)| *id)
     }
 
-    fn on_ready(&mut self) -> anyhow::Result<()> {
+    /// Returns `true` if consensus should be stopped, `false` otherwise.
+    fn on_ready(&mut self) -> anyhow::Result<bool> {
         if !self.node.has_ready() {
-            return Ok(());
+            return Ok(false);
         }
         // Get the `Ready` with `RawNode::ready` interface.
         let ready = self.node.ready();
         let light_rd = self.process_ready(ready)?;
-        self.process_light_ready(light_rd)?;
-        Ok(())
+        if let Some(light_ready) = light_rd {
+            Ok(self.process_light_ready(light_ready)?)
+        } else {
+            // No light ready, so we need to stop consensus.
```

Comment:
I do not understand why we need to stop the consensus here

Reply:
That is just how I decided to propagate this information from `process_ready`. Not sure what would be a better comment here